help_function/src/util_group_load_store_test.cpp

```diff
@@ -0,0 +1,199 @@
+// ====------ onedpl_test_group_load.cpp------------ *- C++ -* ----===//
+
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+#include <dpct/dpct.hpp>
+#include <dpct/dpl_utils.hpp>
+#include <iostream>
+#include <oneapi/dpl/iterator>
+#include <sycl/sycl.hpp>
+
+template <dpct::group::load_algorithm T>
+bool helper_validation_function(const int *ptr, const char *func_name) {
+  if constexpr (T == dpct::group::load_algorithm::BLOCK_LOAD_DIRECT) {
+    for (int i = 0; i < 512; ++i) {
+      if (ptr[i] != i) {
+        std::cout << func_name << "_blocked"
+                  << " failed\n";
+        std::ostream_iterator<int> Iter(std::cout, ", ");
+        std::copy(ptr, ptr + 512, Iter);
+        std::cout << std::endl;
+        return false;
+      }
+    }
+    std::cout << func_name << "_blocked"
+              << " pass\n";
+  } else {
+    for (int i = 0; i < 512; ++i) {
+      if (ptr[i] != i) {
+        std::cout << func_name << "_striped"
+                  << " failed\n";
+        std::ostream_iterator<int> Iter(std::cout, ", ");
+        std::copy(ptr, ptr + 512, Iter);
+        std::cout << std::endl;
+        return false;
+      }
+    }
+    std::cout << func_name << "_striped"
+              << " pass\n";
+  }
+  return true;
+}
+
+bool subgroup_helper_validation_function(const int *ptr, const uint32_t *sg_sz,

```

Comment:
looks like this has the same verification as well (as it should), but it can be consolidated into a single routine.